base/inference.jl

```diff
@@ -5202,6 +5217,21 @@ function occurs_outside_getfield(e::ANY, sym::ANY,
         if head === :(=)
             return occurs_outside_getfield(e.args[2], sym, sv,
                                            field_count, field_names)
+        elseif head === :foreigncall
+            args = e.args
+            nccallargs = args[5]::Int
+            # Only arguments escape, GC root arguments do not escape.
```

Comment:
Yes, adding that comment.

> But also, is that actually true?

It is not ture just from the ccall itself. E.g. I think it is perfectly valid if one write

```julia
function cconvert(....)
    a = Ref(b)
    (a, pointer_from_objref(a))
end
function unsafe_convert(..., v)
    v[2]
end
```

For the ccall, `a` will only appear in the root slot but it cannot be deconstructed since there are escaped use of `a`'s address. OTOH, if we scan through all the uses and there isn't any escaped use, then I think it is valid that we can elide the allocation. I mean, the user is not even using it.....